plans/create.pp

```diff
@@ -0,0 +1,88 @@
+#
+# @summary Plan to provision a Virtual Machine in different cloud providers (GCP, AWS & Azure)
+#
+# @param provider
+#   The cloud provider to use for provisioning
+#
+# @param resource_name
+#   The name of the resource to be provisioned, the same will be use to manage the state of the infrastructure.
+#   So the preference is to use the unit name for the provisioned infrastructure
+#
+# @param instance_type
+#   The instance type to be provisioned, the module will translate the instance type to the cloud provider specific
+#   instance type depending on the provider.
+#
+# @param image
+#   The cloud image that is used for new instance provisioning, format differs
+#   depending on provider
+#
+# @param region
+#   Which region to provision infrastructure in, if not provided default will
+#   be determined by provider
+#
+# @param ssh_key
+#   The SSH key name to be provisioned & access the machine.
```

Comment:
This is unclear. Will the key be provisioned, or should it exist?

Reply:
it should be exist, we will not be provision key.

Reply:
btw updated to `ssh_key_name` and description as current variable name is bit confusing.